lib/cocoapods/target/build_settings.rb

```diff
@@ -623,13 +623,24 @@ def initialize(target, non_library_spec = nil)
 
         # @!group Libraries
 
+        # Converts array of library path references to just the names to use
+        # link each library, e.g. from '/path/to/libSomething.a' to 'Something'
+        #
+        # @param [Array<String>] libraries
+        #
+        # @return [Array<String>]
+        #
+        def linker_names_from_libraries(libraries)
+          libraries.map { |l| File.basename(l, l.extname).sub(/\Alib/, '') }
+        end
+
         # @return [Array<String>]
         define_build_settings_method :libraries, :memoized => true, :sorted => true, :uniqued => true do
           return [] if library_xcconfig? && target.build_as_static?
 
           libraries = []
           if non_library_xcconfig? || target.build_as_dynamic?
-            libraries.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
+            libraries.concat linker_names_from_libraries(vendored_static_libraries)
```

Comment:
`vendored_static_libraries` is correct compared to `vendored_static_frameworks` which was the real fix.